vendor/odoc-parser/src/dune

```diff
@@ -0,0 +1,5 @@
+(ocamllex lexer)
+
+(library
+ (name odoc_parser)
+ (libraries astring result camlp-streams))
```

Comment:
In the dune's vendoring, all the dune files I took inspiration from (such as [this one](https://github.com/ocaml/dune/blob/main/vendor/fmt/src/dune)) have `(wrapped false)`.

I'm not sure if it's needed, but it still compiles without it..

Reply:
It's not needed. It's needed under the following circumstance: the code is being vendored in a library so we need to avoid name collisions with the original library if we link it to the same binary.

In this case, we're using this odoc parser only in ocamllsp, so we can drop it.